scripts/collector/postgres/db-machine-specs.sh

```diff
@@ -41,30 +42,44 @@ echo "$headers" > "$outputPath"
 
 # Only supported locally.
 
-# Check if machineName is "localhost"
-if [ "$machine_name" = "localhost" ]; then
-    machine_name=$(hostname)
-fi
-
-if [ "$machine_name" != "$(hostname)" ]; then    
-    echo "Specified machine_name ${machine_name} does not match the actual hostname. Writing headers only to $outputPath."
-    csvData="\"$pkey\"|\"$dmaSourceId\"|\"$dmaManualId\"|\"$machine_name\"|0|0|0|0|0"
-    echo "$csvData" >> "$outputPath"
-    exit 0
-fi
+coreScript=$(cat <<'EOF'
+    hostName=$(hostname)
+    physicalCpuCount=$(lscpu | awk '/^Socket/{print $2}')
+    logicalCpuCount=$(lscpu | awk '/^Thread/{print $4}')
+    memoryMB=$(free -b | awk '/^Mem/{print ($2+0) / (1024*1024)}')
+    totalSizeBytes=$(df --total / | awk '/total/{printf("%.0f\n", ($2+0) * 1024)}')
+    usedSizeBytes=$(df --output=used -B1 / | awk 'NR==2{printf("%.0f\n", ($1+0))}')
+EOF
+)
 
 # Main script logic
 writeLog "Fetching machine HW specs from computer: $machine_name and storing it in output: $outputPath"
 
-# Get hardware specifications
-physicalCpuCount=$(lscpu | awk '/^Socket/{print $2}')
-logicalCpuCount=$(lscpu | awk '/^Thread/{print $4}')
-memoryMB=$(free -b | awk '/^Mem/{print ($2+0) / (1024*1024)}')
-totalSizeBytes=$(df --total / | awk '/total/{printf("%.0f\n", ($2+0) * 1024)}')
-usedSizeBytes=$(df --output=used -B1 / | awk 'NR==2{printf("%.0f\n", ($1+0))}')
+
+# Check if machineName is "localhost"
+if [ "$machine_name" = "0.0.0.0" ] || grep -q "$machine_name" /etc/hosts; then
+    source <(echo "$coreScript")
+else
+    if [[ -z "$userName" ]]; then
+        echo "VM User name not set, skipping."
+        exit 0
```

Comment:
even in case of failure, i would like to have the CSV populated with the hostname and zeroes.

Reply:
the hostname is the more important part it as we use it

Reply:
Done